### PR TITLE
Feature: Batch voting

### PR DESCRIPTION
**Pull request by fsmanuel**
This is the counterpart to https://github.com/67P/truffle-kredits/pull/26

I can not test it at the moment because the estimated gas doesn't work.
Everything else should be done until we have a better design.

### CODE REVIEW

**Comment by fsmanuel** [comment]
@bumi can you test with this branch. My internet is shitty... and I forgot what to do if the gas estimation fails.

**Comment by bumi** [comment]
will check it out (hopefully later today/evening). 
if the gas estimate fails it either means it can not estimate the gas ;) ... because the contract has an error or I don't know. 
That loop in the batch vote function for sure could be an issue...

**Comment by fsmanuel** [comment]
Good. That sounds like it's your problem not mine 😃

**Comment by bumi** [comment]
:scream:

**Comment by skddc** [comment]
Shouldn't there be a way to suggest a gas amount to the wallet for a transaction? It would seem that this is a basic need for any complex contract interaction, as wallets cannot estimate everything on their own.

**Comment by bumi** [comment]
yes, but the wallet should be able to estimate the gas - "simulating" the execution should be a better estimate than some value that we provide.

**Comment by bumi** [comment]
our issue was related to ganache (+ ethers maye...) 
I first need to review the contract code most of the time if the estimation does not work something is wrong with the contract

**Comment by bumi** [comment]
actually it works well for me! pretty cool actually! 
I did not have to change any gas settings... maybe @skddc can you also check it? 

One issue: 
I have 4 open proposals. I select two, vote for them. These get executed and get removed from the list of open proposals. BUT they stay selected. once I select the remaining proposals and do submit selected it also tries to submit the already submitted ones.

**Comment by fsmanuel** [comment]
Good catch! I’ll remove the selection after the submit!

**Comment by bumi** [comment]
:+1:

we should remove executed proposals also. As maybe...maybe a proposal gets executed before I vote and then my whole batch vote fails. 
It might also make sense to change the behaviour of the vote function and not fail if a proposal is already executed... but just count the vote but don't execute it?

**Comment by bumi** [comment]
we should still merge this one, right? it implements everything needed on batch voting independently of the exact UI. 
the removal of the selection is missing then we're good to merge.

**Comment by skddc** [comment]
I'll check it out when I have everything running on Kovan on my machine.

**Comment by fsmanuel** [comment]
I'll update in a sec.

**Comment by fsmanuel** [comment]
Updated

**Comment by fsmanuel** [comment]
@bumi @skddc can you have another look?

**Comment by skddc** [comment]
Yup, will do this next.

File: app/templates/index.hbs [hunk 1]
```diff
@@ -32,8 +32,7 @@
     <div class="content">
       {{!-- TODO: We need a better naming for kredits.hasAccounts --}}
```
Comment (fsmanuel) [inline app/templates/index.hbs]: That comment should be removed

**Comment by skddc** [comment]
I get the following error: `this.kredits.Operator.functions.batchVote is not a function`

I rebased this on master locally, but that shouldn't change anything I would assume.

**Comment by bumi** [comment]
@skddc that's probably because the ABI of the operator was not up to date: https://github.com/67P/kredits-contracts/commit/7216522d833f11e44377df86b1fde21fecae1e7e

**Comment by bumi** [comment]
I released a new version of kredits-contracts `3.0.2` - can you try with that one?

**Pull request closed without merging**